crates/hyperswitch_domain_models/src/relay.rs

```diff
@@ -0,0 +1,275 @@
+                    currency: relay_refund_request.currency,
+                    reason: relay_refund_request.reason,
+                })
+            }
+        });
+        Self {
+            id: value.id,
+            status: value.status,
+            error,
+            connector_resource_id: value.connector_resource_id,
+            connector_id: value.connector_id,
+            profile_id: value.profile_id,
+            relay_type: value.relay_type,
+            data,
+            connector_reference_id: value.connector_reference_id,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case", untagged)]
+pub enum RelayData {
+    Refund(RelayRefundData),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RelayRefundData {
+    pub amount: MinorUnit,
+    pub currency: enums::Currency,
+    pub reason: Option<String>,
+}
+
+#[derive(Debug)]
+pub enum RelayUpdate {
+    ErrorUpdate {
+        error_code: String,
+        error_message: String,
+        status: enums::RelayStatus,
+    },
+    StatusUpdate {
+        connector_reference_id: Option<String>,
+        status: common_enums::RelayStatus,
+    },
+}
+
+impl From<RelayUpdate> for RelayUpdateInternal {
+    fn from(value: RelayUpdate) -> Self {
+        match value {
+            RelayUpdate::ErrorUpdate {
+                error_code,
+                error_message,
+                status,
+            } => Self {
+                error_code: Some(error_code),
+                error_message: Some(error_message),
+                connector_reference_id: None,
+                status: Some(status),
+                modified_at: common_utils::date_time::now(),
+            },
+            RelayUpdate::StatusUpdate {
+                connector_reference_id,
+                status,
+            } => Self {
+                connector_reference_id,
+                status: Some(status),
+                error_code: None,
+                error_message: None,
+                modified_at: common_utils::date_time::now(),
+            },
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl super::behaviour::Conversion for Relay {
+    type DstType = diesel_models::relay::Relay;
+    type NewDstType = diesel_models::relay::RelayNew;
+
+    async fn convert(self) -> CustomResult<Self::DstType, ValidationError> {
+        Ok(diesel_models::relay::Relay {
+            id: self.id,
+            connector_resource_id: self.connector_resource_id,
+            connector_id: self.connector_id,
+            profile_id: self.profile_id,
+            merchant_id: self.merchant_id,
+            relay_type: self.relay_type,
+            request_data: self
+                .request_data
+                .map(|data| {
+                    serde_json::to_value(data).change_context(ValidationError::InvalidValue {
+                        message: "Failed while decrypting business profile data".to_string(),
```

Comment:
If we're still going to use `serde_json::Value` on the diesel model, then update the error message. Otherwise, update the code as required.